agent/asm/asm_test.go

```diff
@@ -111,11 +121,74 @@ func TestASMGetAuthConfig(t *testing.T) {
 }
 
 func TestGetSecretFromASM(t *testing.T) {
-	asmClient := mockGetSecretValue{
-		Resp: secretsmanager.GetSecretValueOutput{
-			SecretString: aws.String("secretValue"),
-		},
-	}
+	asmClient := createASMInterface(secretValue)
 	_, err := GetSecretFromASM("secretName", asmClient)
 	assert.NoError(t, err)
 }
+
+func TestGetSecretFromASMWithJsonKey(t *testing.T) {
+	asmClient := createASMInterface(jsonSecretValue)
+	secretValueInput := createSecretValueInput(toPtr(valueFrom), nil, nil)
```

Comment:
You don't really need toPtr here. You can just update createSecretValueInput function to accept string and use aws.String(XX) to construct the input inside that function. Same below.

Reply:
it was bc i wanted to pass in nil for versionID or versionStage.